test/compress/destructuring.js

```diff
@@ -639,3 +639,25 @@ issue_2044_ecma_6_beautify: {
     }
     expect_exact: "({x: a = 1, y = 2 + b, z = 3 - c} = obj);"
 }
+
+issue_2140: {
+    options = {
+        unused: true,
+    }
+    input: {
+        !function() {
+            var s = {}, t = {};
+            [t.a] = [42];
+            console.log(t.a);
```

Comment:
Prior to this PR the extra reference to `t.a` seems to have inadvertantly "fixed" the test case above. Or maybe my tree is out of sync.